File: js/stage1/S1_VAR_10.js
// S1_VAR_10 — Mini debugger
function inspect(value) {
  return {
    type: typeof value,
    isArray: Array.isArray(value),
    isNull: value === null,
    isNaN: Number.isNaN(value),
  };
}
[null, undefined, 0, NaN, [], {}, "x", () => {}].forEach((v) =>
  console.log(inspect(v))
);

File: Python/stage1/S1_VAR_10.py
# S1_VAR_10 — Mini inspector
def inspect(value):
   return {
       "type": type(value),
       "is_none": value is None,
       "is_list": isinstance(value, list),
       "is_bool": isinstance(value, bool),
   }
tests = [None, 0, True, [], {}, "hi"]
for t in tests:
   print(inspect(t))